limacharlie/artifact.go

```diff
@@ -76,6 +86,48 @@ func (org Organization) ArtifactRuleDelete(ruleName ArtifactRuleName) error {
 	return nil
 }
 
+func (org Organization) GetUploadUrl(oid string) (string, error) {
+
+	resp := orgUrls{}
+	var request restRequest
+	request = makeDefaultRequest(&resp)
+
+	if err := org.client.reliableRequest(http.MethodGet, fmt.Sprintf("orgs/%s/url", org.GetOID()), request); err != nil {
+		fmt.Println("Error making GET request:", err)
+		return "", nil
+	}
+
+	logsUrl := resp.URL
+	return logsUrl["logs"], nil
+}
+
+func (org Organization) CreateArtifactFromBytes(name ArtifactName, data []byte) error {
+	return org.CreateArtifactFromReader(name, bytes.NewBuffer(data))
+}
+
+func (org Organization) CreateArtifactFromReader(name ArtifactName, data io.Reader) error {
+	resp := artifactPutPointer{}
+	orgUrl, _ := org.GetUploadUrl(org.GetOID())
+	request := makeDefaultRequest(&resp)
+	if err := org.client.reliableRequest(http.MethodPost, fmt.Sprintf("https://%s/ingest", &orgUrl), request); err != nil {
+		return err
+	}
+	c := &http.Client{}
+	req, err := http.NewRequest(http.MethodPut, resp.URL, data)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/octet-stream")
+	httpResp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+	if httpResp.StatusCode != 200 {
+		return fmt.Errorf("failed to PUT artifact, http status: %d", httpResp.StatusCode)
+	}
+	return nil
```

Comment:
Sadly this is not how the ingestion works.
It expects a series of POST in chunks of 10MB with a few authentication headers: https://github.com/refractionPOINT/python-limacharlie/blob/master/limacharlie/Logs.py#L96

Reply:
dammit. i had started that way and saw this in payloads and thought i had it backwards. will revert.